Sources/BraintreeCore/BTAppContextSwitcher.m

```diff
@@ -44,18 +44,18 @@ + (BOOL)handleOpenURLContext:(UIOpenURLContext *)URLContext {
 }
 
 - (BOOL)handleOpenURL:(NSURL *)url {
-    for (Class<BTAppContextSwitchDriver> driverClass in self.appContextSwitchDrivers) {
-        if ([driverClass canHandleReturnURL:url]) {
-            [driverClass handleReturnURL:url];
+    for (Class<BTAppContextSwitchClient> clientClass in self.appContextSwitchClients) {
+        if ([clientClass canHandleReturnURL:url]) {
+            [clientClass handleReturnURL:url];
             return YES;
         }
     }
     return NO;
 }
 
-- (void)registerAppContextSwitchDriver:(Class<BTAppContextSwitchDriver>)driver {
-    if (!driver) return;
-    [self.appContextSwitchDrivers addObject:driver];
+- (void)registerAppContextSwitchClient:(Class<BTAppContextSwitchClient>)client {
+    if (!client) return;
+    [self.appContextSwitchClients addObject:client];
```

Comment:
This is unrelated but I wonder if we've been reinventing [NotificationCenter](https://developer.apple.com/documentation/foundation/notificationcenter) to a degree here.

Reply:
Yeah we basically are. 🙃 My _guess_ is that as we start rewriting core we will see a similar pattern and make a decision on either keeping it (unlikely) or move to something like NotificationCenter, then when we get back to it here we will hopefully be able to carry on with some new pattern.